src/Jackett.Common/Indexers/TorrentParadise.cs

```diff
@@ -0,0 +1,167 @@
+            public TorrentParadiseResult(string id, string text, string len, string s, string l)
+            {
+                Id = id;
+                Text = text;
+                Size = ToLong(len);
+                Seeders = ToLong(s);
+                Leechers = ToLong(l);
+            }
+
+            public string Id { get; }
+
+            public string Text { get; }
+
+            public long? Size { get; }
+
+            public long? Seeders { get; }
+
+            public long? Leechers { get; }
+
+            private long? ToLong(string str) => Convert.ToInt64(str);
+        }
+
+        private string ApiUrl => SiteLink + "api/";
+
+        public TorrentParadise(IIndexerConfigurationService configService, WebClient wc, Logger l, IProtectionService ps,
+            ICacheService cs)
+            : base(id: "torrent-paradise",
+                   name: "Torrent Paradise",
+                   description: "The most innovative torrent site",
+                   link: "https://torrent-paradise.ml/",
+                   caps: new TorznabCapabilities
+                   {
+                       TvSearchParams = new List<TvSearchParam>
+                       {
+                           TvSearchParam.Q, TvSearchParam.Season, TvSearchParam.Ep
+                       },
+                       MovieSearchParams = new List<MovieSearchParam>
+                       {
+                           MovieSearchParam.Q
+                       },
+                       MusicSearchParams = new List<MusicSearchParam>
+                       {
+                           MusicSearchParam.Q
+                       },
+                       BookSearchParams = new List<BookSearchParam>
+                       {
+                           BookSearchParam.Q
+                       }
+                   },
+                   configService: configService,
+                   client: wc,
+                   logger: l,
+                   p: ps,
+                   cacheService: cs,
+                   configData: new ConfigurationData())
+        {
+            Encoding = Encoding.GetEncoding("UTF-8");
+            Language = "en-us";
+            Type = "public";
+
+            AddCategoryMapping(8000, TorznabCatType.Other);
+        }
+
+        public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
+        {
+            LoadValuesFromJson(configJson);
+
+            var query = new TorznabQuery { SearchTerm = DateTime.Now.Year.ToString(), IsTest = true };
+            var releases = await PerformQuery(query);
+
+            await ConfigureIfOK(string.Empty, releases.Any(),
+                                () => throw new Exception("Could not find releases from this URL"));
+
+            return IndexerConfigurationStatus.Completed;
+        }
+
+        protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
+        {
+            var searchUrl = CreateSearchUrl(query);
```

Comment:
Remove the CreateSearchUrl method and use this code instead (not tested):
```
var searchTerm = query.GetQueryString();
if (string.IsNullOrWhiteSpace(searchTerm)) {
   if (query.IsTest)
         searchTerm = DateTime.Now.Year.ToString();
   else 
         return new List<ReleaseInfo>();
}
 var qc = new List<KeyValuePair<string, string>>
{
    {"q", searchTerm}
};
var searchUrl = ApiUrl + "search?" + qc.GetQueryString();
```

Reply:
I'd like to keep it as a method since it does its own specific task and it also keeps `PerformQuery` less cluttered.
I've changed it a bit, as you suggested. I still update the `query.SearchTerm`, which keeps everything in sync (for when you might use `TorznabQuery query` later, in `PerformQuery`).

Reply:
@XYZJR There are few things I didn't tell you. You can keep the method, it's fine.
1) You shouldn't modify the query object because it's shared by other indexers. In Jackett you can make a search in several indexers at the same time and all of them receive the same query object. This is a bug in Jackett. I tried to fix it in this PR but I think it should be a better fix based on immutable objects. Maybe you can take a look => #8570
2) Replace `List<KeyValuePair<string, string>>` with `NameValueCollection`.  My bad. We are using this object to concatenate query parameters because `qc.GetQueryString()` escapes the keys and values to form a valid URL

Reply:
Alright, changing it now.

PS: Is there an easier way to get in contact with you, are you on the Jackett Discord for example?

Reply:
email ngosang at hotmail.es but the time to response is similar to github.
I will try to review your PRs faster but I'm very busy, sorry.